Wrong path to nodeDebug.js

Hi,

trying to run a Node.js application in eclipse I get the following error (note the double `C:\`):
```
module.js:549
    throw err;
    ^

Error: Cannot find module 'C:\C:\eclipse\plugins\org.eclipse.wildwebdeveloper_0.3.0.201906060852\language-servers\node_modules\node-debug2\out\src\nodeDebug.js'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```
Looking at the corresponding launch configuration there is the entry `listEntry`:
```
<launchConfiguration ...>
...
    <listAttribute key="org.eclipse.lsp4e.debug.model.ATTR_DSP_ARGS">
      <listEntry value="/C:/eclipse/plugins/org.eclipse.wildwebdeveloper_0.3.0.201906060852/language-servers/node_modules/node-debug2/out/src/nodeDebug.js"/>
...
    </listAttribute>
</launchConfiguration>
```
My guess is the the leading slash causing the double `C:\`. 

In the classes files NodeAttachDebugDelegate (in line 56) and NodeRunDAPDebugDelegate (in line 86) I've found the following line:
```
wc.setAttribute(DSPPlugin.ATTR_DSP_ARGS, Collections.singletonList(FileLocator.toFileURL(getClass().getResource("/language-servers/node_modules/node-debug2/out/src/nodeDebug.js")).getPath())); //$NON-NLS-1$
```
Maybe the used `org.eclipse.core.runtime.FileLocator` or org.eclipse.core.internal.runtime.Activator.getURLConverter() causing this behavior.

Is there a possible workaround by setting an environment variable?

Installation details:
- Windows 10
- Eclipse IDE Version 2019-06 (4.12.0)
- Wild Web Developer - Web edition in Eclipse IDE (Experimental)	0.4.1.201906280859	org.eclipse.wildwebdeveloper.feature.feature.group	Eclipse Wild Web Developer project

Kind regards

Sven

--- a/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/InitializeLaunchConfigurations.java
+++ b/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/InitializeLaunchConfigurations.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2016 Red Hat Inc. and others.
+ * Copyright (c) 2019 Red Hat Inc. and others.
  *
  * This program and the accompanying materials are made
  * available under the terms of the Eclipse Public License 2.0
@@ -10,6 +10,7 @@
  * Contributors:
  *   Mickael Istria (Red Hat Inc.) - initial implementation
  *   Gautier de Saint Martin Lacaze - Issue #55 Warn missing or incompatible node.js
+ *   Pierre-Yves B. - Issue #196 NullPointerException when validating Node.js version
  *******************************************************************************/
 package org.eclipse.wildwebdeveloper;
 

--- a/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/NodeAttachDebugDelegate.java
+++ b/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/NodeAttachDebugDelegate.java
@@ -9,9 +9,11 @@
  *
  * Contributors:
  *   Mickael Istria (Red Hat Inc.) - initial implementation
+ *   Pierre-Yves B. - Issue #180 Wrong path to nodeDebug.js
  *******************************************************************************/
 package org.eclipse.wildwebdeveloper.debug;
 
+import java.io.File;
 import java.io.IOException;
 import java.net.URL;
 import java.util.Collections;
@@ -50,7 +52,8 @@ public class NodeAttachDebugDelegate extends DSPLaunchDelegate {
 		try {
 			URL fileURL = FileLocator.toFileURL(
 					getClass().getResource("/language-servers/node_modules/node-debug2/out/src/nodeDebug.js"));
-			List<String> debugCmdArgs = Collections.singletonList(fileURL.getPath());
+			File file = new File(fileURL.getPath());
+			List<String> debugCmdArgs = Collections.singletonList(file.getAbsolutePath());
 
 			DSPLaunchDelegateLaunchBuilder builder = new DSPLaunchDelegateLaunchBuilder(configuration, mode, launch,
 					monitor);

--- a/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/NodeRunDAPDebugDelegate.java
+++ b/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/NodeRunDAPDebugDelegate.java
@@ -9,9 +9,11 @@
  *
  * Contributors:
  *   Mickael Istria (Red Hat Inc.) - initial implementation
+ *   Pierre-Yves B. - Issue #180 Wrong path to nodeDebug.js
  *******************************************************************************/
 package org.eclipse.wildwebdeveloper.debug;
 
+import java.io.File;
 import java.io.IOException;
 import java.net.URL;
 import java.util.Arrays;
@@ -76,7 +78,8 @@ public class NodeRunDAPDebugDelegate extends DSPLaunchDelegate {
 		try {
 			URL fileURL = FileLocator.toFileURL(
 					getClass().getResource("/language-servers/node_modules/node-debug2/out/src/nodeDebug.js"));
-			List<String> debugCmdArgs = Collections.singletonList(fileURL.getPath());
+			File file = new File(fileURL.getPath());
+			List<String> debugCmdArgs = Collections.singletonList(file.getAbsolutePath());
 
 			DSPLaunchDelegateLaunchBuilder builder = new DSPLaunchDelegateLaunchBuilder(configuration, mode, launch,
 					monitor);

--- a/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/jsts/TypeScriptPlugin.java
+++ b/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/jsts/TypeScriptPlugin.java
@@ -6,10 +6,15 @@
  * which is available at https://www.eclipse.org/legal/epl-2.0/
  *
  * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Pierre-Yves B. - Issue #180 Wrong path to nodeDebug.js
  *******************************************************************************/
 package org.eclipse.wildwebdeveloper.jsts;
 
+import java.io.File;
 import java.io.IOException;
+import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -21,7 +26,8 @@ public class TypeScriptPlugin {
 	
 	public TypeScriptPlugin(String name) throws IOException {
 		pluginName = name;
-		pluginProbeLocation = FileLocator.toFileURL(getClass().getResource("/language-servers/node_modules/" + name)).toString();
+		URL fileURL = FileLocator.toFileURL(getClass().getResource("/language-servers/node_modules/" + name));
+		pluginProbeLocation = new File(fileURL.getPath()).getAbsolutePath();
 	}
 	
 	public Map<String, String> toMap() {

